### PR TITLE
Move extension files to subfolder

### PR DESCRIPTION
**Pull request by JatinSanghvi**
- Keeping the root directory clean. Motivation for changes:
  - Java and .NET coding require separate environments for coding. For instance, if one has opened the root folder in VS code, they will in any case be unable to build and debug Java library code.
  - Commands like `dotnet test` will see a .NET project structure, The CLI won't find Java files in first place to be able to ignore them.

- Fixed regex string used for parsing version tag. It was not allowing a version tag like `v2.0.0`

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -22,7 +22,7 @@
     │       ├───Microsoft.Azure.WebJobs.Extensions.RabbitMQ.2.0.0-preview.symbols.nupkg
     │       └───_manifest
     │           └───<files>
-    └───drop-java-bindings
+    └───drop-java-library
         └───2.0.0-preview
             ├───_manifest
             │   └───<files>
```
Comment (JatinSanghvi) [inline RELEASE_PROCESS.md:28]: Modified three files: `RELEASE_PROCESS.md`, `azure-pipeline.yml`, and `.gitignore`. Rest of all files are only shifted to different folders.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
     displayName: WebJobs extension
     steps:
       - powershell: | # Allow tags matching v1.2.3 and v1.2.3-xyz1
-          $found = '$(Build.SourceBranchName)' | Select-String -Pattern '^v((\d+\.\d+\.\d+)(?:-\w+))*$'
+          $found = '$(Build.SourceBranchName)' | Select-String -Pattern '^v((?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-\w+)?)$'
           if (-not $found) {
```
Comment (JatinSanghvi) [inline azure-pipelines.yml:33]: The earlier pattern was incorrect (don't remember why I even committed it like this 🤦). It did not allow a tag like `v2.0.0` which is intended tag for next major release.

**Review by v-shenoy — APPROVED** [review]

**Comment by JatinSanghvi** [comment]
Thanks for approving the PR, @v-shenoy. I found two issues with the build pipeline that impacted release builds. Please check the [latest commit](https://github.com/Azure/azure-functions-rabbitmq-extension/pull/189/commits/265f6f18a8a7612f350ac5023d49e2d98e8ce428) at your free time and approve the PR if things look good.

File: azure-pipelines.yml (hunk 2 above)
Comment (v-shenoy) [reply azure-pipelines.yml:33]: I should have also look at it it carefully the previous time, my bad.

**Review by v-shenoy — APPROVED** [review]

**Comment by v-shenoy** [comment]
> Thanks for approving the PR, @v-shenoy. I found two issues with the build pipeline that impacted release builds. Please check the [latest commit](https://github.com/Azure/azure-functions-rabbitmq-extension/pull/189/commits/265f6f18a8a7612f350ac5023d49e2d98e8ce428) at your free time and approve the PR if things look good.

Approved, @JatinSanghvi

File: azure-pipelines.yml (hunk 2 above)
Comment (JatinSanghvi) [reply azure-pipelines.yml:33]: I don't think you reviewed the PR with this particular change :)

Comment (JatinSanghvi) [reply azure-pipelines.yml:33]: I had added above PR comment soon after the making the first commit. I meant I should have checked the regex when I had worked on build pipeline 1-2 months back.

**Pull request merged**